products/views.py

```diff
@@ -1,54 +1,68 @@
+from django.views.generic.list import ListView
 from django.shortcuts import render, get_object_or_404
-from django.core.paginator import Paginator
 from django.db.models.functions import Lower
+from django.views.generic import DetailView
 from django_extensions.db.models import ActivatorModel
 
 from .models import Product, Category, Vendor, ProductSpecificationCategory
 
 
-def product_detail(request, product_id, category_name):
-    selected_product = get_object_or_404(Product, id=product_id, status=ActivatorModel.ACTIVE_STATUS)
-    spec_groups = ProductSpecificationCategory.objects.filter(product=selected_product)
+class ProductDetailsView(DetailView):
+    model = Product
+    template_name = "product_detail.html"
+    context_object_name = "selected_product"
 
-    return render(request, "product_detail.html", {
-        "selected_product": selected_product,
-        "spec_groups": spec_groups,
-    })
+    def get_object(self, queryset=None):
+        selected_product = get_object_or_404(
+            Product,
+            id=self.kwargs["product_id"],
+            status=ActivatorModel.ACTIVE_STATUS
+        )
+        return selected_product
```

Comment:
we can do this without overriding get_object method
for your reference: https://docs.djangoproject.com/en/5.1/ref/class-based-views/mixins-single-object/#singleobjectmixin